src/protobuf-net.Reflection/Internal/CodeGen/CodeGenCSharpCodeGenerator.cs

```diff
@@ -151,7 +151,16 @@ protected override void WriteFileHeader(CodeGenGeneratorContext ctx, CodeGenFile
     /// <inheritdoc/>
     protected override void WriteNamespaceHeader(CodeGenGeneratorContext ctx, string @namespace)
     {
-        ctx.WriteLine($"namespace {@namespace}");
+        if (!string.IsNullOrEmpty(@namespace))
```

Comment:
`.TrimEnd('.')` (don't need to test anything then) maybe?

Reply:
refactored